### ISSUE TITLE
Compiling gRPC on PHP 8.0 throws errors

### ISSUE THREAD

**Issue by stemis** [grpc/grpc#24743]
### What version of gRPC and what language are you using?
**PHP**
```
PHP 8.0.0RC3 (cli) (built: Nov  5 2020 19:24:54) ( NTS )
Copyright (c) The PHP Group
Zend Engine v4.0.0-dev, Copyright (c) Zend Technologies
```
**gRPC**
Version: 1.33.1

### What operating system (Linux, Windows,...) and version?
Docker: `php:8.0-rc-cli`
```
PRETTY_NAME="Debian GNU/Linux 10 (buster)"
NAME="Debian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
```
### What runtime / compiler are you using (e.g. python version or version of gcc)
```
Using built-in specs.
COLLECT_GCC=gcc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/8/lto-wrapper
OFFLOAD_TARGET_NAMES=nvptx-none
OFFLOAD_TARGET_DEFAULT=1
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 8.3.0-6' --with-bugurl=file:///usr/share/doc/gcc-8/README.Bugs --enable-languages=c,ada,c++,go,brig,d,fortran,objc,obj-c++ --prefix=/usr --with-gcc-major-version-only --program-suffix=-8 --program-prefix=x86_64-linux-gnu- --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without
-included-gettext --enable-threads=posix --libdir=/usr/lib --enable-nls --enable-bootstrap --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --with-default-libstdcxx-abi=new --enable-gnu-unique-object --disable-vtable-verify --enable-libmpx --enable-plugin --enable-default-pie --with-system-zlib --with-target-system-zlib --enable-o
bjc-gc=auto --enable-multiarch --disable-werror --with-arch-32=i686 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-offload-targets=nvptx-none --without-cuda-driver --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 8.3.0 (Debian 8.3.0-6)
```

### What did you do?
```
wget https://github.com/FriendsOfPHP/pickle/releases/download/v0.6.0/pickle.phar
mv pickle.phar /usr/local/bin/pickle
chmod +x /usr/local/bin/pickle
pickle install grpc
```

### What did you expect to see?
It succeeding

### What did you see instead?
It failing

```
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/channel_credentials.h:44:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_init_channel_credentials(TSRMLS_D);
2:  ^~~~
2: In file included from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:24,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:24:
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/channel.h:56:30: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:                               TSRMLS_DC);
2:                               ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/channel.h:61:44: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:                                             TSRMLS_DC);
2:                                             ^~~~~~~~~
2: In file included from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:24:
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:44:69: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:  zval *grpc_parse_metadata_array(grpc_metadata_array *metadata_array TSRMLS_DC);
2:                                                                      ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:47:57: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:  zval *grpc_php_wrap_call(grpc_call *wrapped, bool owned TSRMLS_DC);
2:                                                          ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:50:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_init_call(TSRMLS_D);
2:  ^~~~
2: In file included from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:31:
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call_credentials.h:59:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_init_call_credentials(TSRMLS_D);
2:  ^~~~
2: In file included from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:32:
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/completion_queue.h:30:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_php_init_completion_queue(TSRMLS_D);
2:  ^~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/completion_queue.h:33:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_php_shutdown_completion_queue(TSRMLS_D);
2:  ^~~~
2: In file included from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:33:
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/timeval.h:39:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_init_timeval(TSRMLS_D);
2:  ^~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/timeval.h:42:1: error: parameter names (without types) in function declaration [-Werror]
2:  void grpc_shutdown_timeval(TSRMLS_D);
2:  ^~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/timeval.h:45:50: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:  zval *grpc_php_wrap_timeval(gpr_timespec wrapped TSRMLS_DC);
2:                                                   ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:50:47: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:                                                TSRMLS_DC) {
2:                                                ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:60:49: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:                                  *metadata_array TSRMLS_DC) {
2:                                                  ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:184:57: error: expected ';', ',' or ')' before 'TSRMLS_DC'
2:  zval *grpc_php_wrap_call(grpc_call *wrapped, bool owned TSRMLS_DC) {
2:                                                          ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c: In function 'zim_Call___construct':
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:214:45: error: expected ')' before 'TSRMLS_CC'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "OsO|s", &channel_obj,
2:                                              ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:214:7: error: too few arguments to function 'zend_parse_parameters'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "OsO|s", &channel_obj,
2:        ^~~~~~~~~~~~~~~~~~~~~
2: In file included from /usr/local/include/php/main/php.h:35,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/php_grpc.h:29,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:22,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:24:
2: /usr/local/include/php/Zend/zend_API.h:304:22: note: declared here
2:  ZEND_API zend_result zend_parse_parameters(uint32_t num_args, const char *type_spec, ...);
2:                       ^~~~~~~~~~~~~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:220:49: error: expected ')' before 'TSRMLS_CC'
2:                           "an optional String", 1 TSRMLS_CC);
2:                                                  ^~~~~~~~~~
2:                                                  )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:228:27: error: expected ')' before 'TSRMLS_CC'
2:                           1 TSRMLS_CC);
2:                            ^~~~~~~~~~
2:                            )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:235:27: error: expected ')' before 'TSRMLS_CC'
2:                           1 TSRMLS_CC);
2:                            ^~~~~~~~~~
2:                            )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:209:16: error: unused variable 'host_override_len' [-Werror=unused-variable]
2:    php_grpc_int host_override_len = 0;
2:                 ^~~~~~~~~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:206:16: error: unused variable 'method_len' [-Werror=unused-variable]
2:    php_grpc_int method_len;
2:                 ^~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c: In function 'zim_Call_startBatch':
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:281:29: error: expected ')' before 'TSRMLS_CC'
2:                             1 TSRMLS_CC);
2:                              ^~~~~~~~~~
2:                              )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:317:45: error: expected ')' before 'TSRMLS_CC'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "a", &array) ==
2:                                              ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:317:7: error: too few arguments to function 'zend_parse_parameters'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "a", &array) ==
2:        ^~~~~~~~~~~~~~~~~~~~~
2: In file included from /usr/local/include/php/main/php.h:35,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/php_grpc.h:29,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:22,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:24:
2: /usr/local/include/php/Zend/zend_API.h:304:22: note: declared here
2:  ZEND_API zend_result zend_parse_parameters(uint32_t num_args, const char *type_spec, ...);
2:                       ^~~~~~~~~~~~~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:320:59: error: expected ')' before 'TSRMLS_CC'
2:                           "start_batch expects an array", 1 TSRMLS_CC);
2:                                                            ^~~~~~~~~~
2:                                                            )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:332:60: error: expected ')' before 'TSRMLS_CC'
2:                             "batch keys must be integers", 1 TSRMLS_CC);
2:                                                             ^~~~~~~~~~
2:                                                             )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:344:59: error: expected ')' before 'TSRMLS_CC'
2:                               "Bad metadata value given", 1 TSRMLS_CC);
2:                                                            ^~~~~~~~~~
2:                                                            )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:354:31: error: expected ')' before 'TSRMLS_CC'
2:                               1 TSRMLS_CC);
2:                                ^~~~~~~~~~
2:                                )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:363:33: error: expected ')' before 'TSRMLS_CC'
2:                                 1 TSRMLS_CC);
2:                                  ^~~~~~~~~~
2:                                  )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:372:31: error: expected ')' before 'TSRMLS_CC'
2:                               1 TSRMLS_CC);
2:                                ^~~~~~~~~~
2:                                )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:388:33: error: expected ')' before 'TSRMLS_CC'
2:                                 1 TSRMLS_CC);
2:                                  ^~~~~~~~~~
2:                                  )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:401:33: error: expected ')' before 'TSRMLS_CC'
2:                                 1 TSRMLS_CC);
2:                                  ^~~~~~~~~~
2:                                  )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:409:31: error: expected ')' before 'TSRMLS_CC'
2:                               1 TSRMLS_CC);
2:                                ^~~~~~~~~~
2:                                )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:417:33: error: expected ')' before 'TSRMLS_CC'
2:                                 1 TSRMLS_CC);
2:                                  ^~~~~~~~~~
2:                                  )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:427:31: error: expected ')' before 'TSRMLS_CC'
2:                               1 TSRMLS_CC);
2:                                ^~~~~~~~~~
2:                                )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:450:58: error: expected ')' before 'TSRMLS_CC'
2:                             "Unrecognized key in batch", 1 TSRMLS_CC);
2:                                                           ^~~~~~~~~~
2:                                                           )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:461:37: error: expected ')' before 'TSRMLS_CC'
2:                           (long)error TSRMLS_CC);
2:                                      ^~~~~~~~~~
2:                                      )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:488:17: error: implicit declaration of function 'grpc_parse_metadata_array'; did you mean 'create_metadata_array'? [-Werror=implicit-function-declaration]
2:        recv_md = grpc_parse_metadata_array(&recv_metadata);
2:                  ^~~~~~~~~~~~~~~~~~~~~~~~~
2:                  create_metadata_array
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:488:15: error: assignment to 'zval *' {aka 'struct _zval_struct *'} from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
2:        recv_md = grpc_parse_metadata_array(&recv_metadata);
2:                ^
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:527:15: error: assignment to 'zval *' {aka 'struct _zval_struct *'} from 'int' makes pointer from integer without a cast [-Werror=int-conversion]
2:        recv_md = grpc_parse_metadata_array(&recv_trailing_metadata);
2:                ^
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c: In function 'zim_Call_setCredentials':
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:606:45: error: expected ')' before 'TSRMLS_CC'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &creds_obj,
2:                                              ^~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:606:7: error: too few arguments to function 'zend_parse_parameters'
2:    if (zend_parse_parameters(ZEND_NUM_ARGS() TSRMLS_CC, "O", &creds_obj,
2:        ^~~~~~~~~~~~~~~~~~~~~
2: In file included from /usr/local/include/php/main/php.h:35,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/php_grpc.h:29,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.h:22,
2:                  from /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:24:
2: /usr/local/include/php/Zend/zend_API.h:304:22: note: declared here
2:  ZEND_API zend_result zend_parse_parameters(uint32_t num_args, const char *type_spec, ...);
2:                       ^~~~~~~~~~~~~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:610:27: error: expected ')' before 'TSRMLS_CC'
2:                           1 TSRMLS_CC);
2:                            ^~~~~~~~~~
2:                            )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c: In function 'grpc_init_call':
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:654:6: error: type of 'TSRMLS_D' defaults to 'int' [-Werror=implicit-int]
2:  void grpc_init_call(TSRMLS_D) {
2:       ^~~~~~~~~~~~~~
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:657:22: error: 'create_wrapped_grpc_call' undeclared (first use in this function); did you mean 'free_wrapped_grpc_call'?
2:    ce.create_object = create_wrapped_grpc_call;
2:                       ^~~~~~~~~~~~~~~~~~~~~~~~
2:                       free_wrapped_grpc_call
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:657:22: note: each undeclared identifier is reported only once for each function it appears in
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c:658:50: error: expected ')' before 'TSRMLS_CC'
2:    grpc_ce_call = zend_register_internal_class(&ce TSRMLS_CC);
2:                                                   ^~~~~~~~~~
2:                                                   )
2: /tmp/grpc/grpc-1.33.1/src/php/ext/grpc/call.c: At top level:
2: cc1: error: unrecognized command line option '-Wno-parentheses-equality' [-Werror]
2: cc1: all warnings being treated as errors
2: make: *** [Makefile:1169: src/php/ext/grpc/call.lo] Error 1
```
See attached full gcc log [gcc.log](https://github.com/grpc/grpc/files/5536970/gcc.log)
### Anything else we should know about your project / environment?
No

**Comment by stanley-cheung** [comment]
This is being worked on right now.

**Comment by stanley-cheung** [comment]
PHP 8 is now supported in the latest release candidate `1.34.0RC1`. You can try it out by doing `sudo pecl install grpc-beta`.

**Comment by stemis** [comment]
> PHP 8 is now supported in the latest release candidate `1.34.0RC1`. You can try it out by doing `sudo pecl install grpc-beta`.

Thanks!

@stanley-cheung 
I am running into a few issues since version `1.34.0RC1` is not [SemVer](https://semver.org/). Could you change this to `1.34.0-RC1` ?

**Comment by stanley-cheung** [comment]
Hm, I am not sure where the disconnect is. My understanding is that you either do `sudo pecl install grpc-beta` or `sudo pecl install grpc-1.34.0RC1`, but never `1.34.0RC1-beta`. On the other hand, we had been naming the release candidate similar to `1.34.0RC1` for a while now, and is consistent with some other pecl extensions like [xdebug](http://pecl.php.net/package/xdebug). In about 2 weeks the final released version will be `1.34.0` which should be fine though.

**Comment by stemis** [comment]
You are correct, there was an issue with the `pickle` package I was using...
I can confirm gRPC is now compiling correctly on php-8.0-RC!
